test/unit/db/repos/blocks.js

```diff
@@ -0,0 +1,885 @@
+
+				expect(db.query.firstCall.args[0]).to.eql(sql.getHeightByLastId);
+				expect(db.query.firstCall.args[1]).to.eql(['11111']);
+				return expect(db.query).to.be.calledOnce;
+			});
+
+			it('should be fulfilled with empty response if required param "lastId" is missing', () => {
+				return expect(db.blocks.getHeightByLastId()).to.be.eventually.eql([]);
+			});
+
+			it('should correct height of the given block id', function*() {
+				const genesisBlock = blocksFixtures.GenesisBlock();
+				const height = yield db.blocks.getHeightByLastId(genesisBlock.id);
+
+				expect(height).to.be.not.empty;
+				expect(height).to.be.an('array');
+				expect(height).to.be.lengthOf(1);
+				expect(height[0]).to.have.all.keys('height');
+				return expect(height[0].height).to.be.eql(genesisBlock.height);
+			});
+		});
+
+		describe('loadBlocksData()', () => {
+			it('should use the correct SQL file with no parameters', function*() {
+				sinonSandbox.spy(db, 'query');
+
+				const params = {
+					id: '1111',
+					lastId: '1111',
+					height: 1,
+					limit: 10,
+				};
+				yield db.blocks.loadBlocksData(params);
+				expect(db.query.firstCall.args[1]).to.eql(params);
+				return expect(db.query).to.be.calledOnce;
+			});
+
+			it('should be rejected if required param "limit" is missing', () => {
+				return expect(
+					db.blocks.loadBlocksData({
+						id: '1111',
+						lastId: '1111',
+						height: 1,
+					})
+				).to.be.eventually.rejectedWith("Property 'limit' doesn't exist.");
+			});
+
+			it('should be rejected if required param "height" is missing', () => {
+				return expect(
+					db.blocks.loadBlocksData({
+						id: '1111',
+						lastId: '1111',
+						limit: 10,
+					})
+				).to.be.eventually.rejectedWith("Property 'height' doesn't exist.");
+			});
+
+			it('should return data for given block id', function*() {
+				const block = blocksFixtures.GenesisBlock();
+				const data = yield db.blocks.loadBlocksData({
+					id: block.id,
+					limit: 10,
+				});
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('array');
+				expect(data).to.be.lengthOf(1);
+				expect(data[0].b_id).to.be.eql(block.id);
+				return expect(data[0]).to.have.all.keys(blockListFields);
+			});
+
+			it('should return data for given lastBlock id', function*() {
+				const block = blocksFixtures.GenesisBlock();
+				const data = yield db.blocks.loadBlocksData({
+					lastId: block.id,
+					height: 1,
+					limit: 10,
+				});
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('array');
+				// There are 5 total blocks in the seed
+				return expect(data).to.be.lengthOf(4);
+			});
+
+			it('should use condition limit condition as height if both "id" and "lastId" is not provided', function*() {
+				const data = yield db.blocks.loadBlocksData({
+					height: 1,
+					limit: 3,
+				});
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('array');
+				// There are 5 total blocks in the seed
```

Comment:
same here, please improve that comment - query: 3, total: 5, expect: 2 - it need explanation ;)